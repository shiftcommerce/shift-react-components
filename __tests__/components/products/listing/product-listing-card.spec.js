// Libraries
import React from 'react'

// Components
import ProductListingCard from '../../../../src/components/products/listing/product-listing-card'

// Fixtures
import productSearchHit from '../../../fixtures/product-search-hit'

test('renders ProductListingCard correctly', () => {
  // Act

  // at the moment we dont pass the image size throught algolia,
  // the fixture will need updating when we do. At the moment
  // product listing is set to default to 280

  const wrapper = mount(
    <ProductListingCard
      title={productSearchHit.product_title}
      assetFileUrl={productSearchHit.product_assets[0].url}
      assetFileAltText={productSearchHit.product_assets[0].alt_text}
      minPrice={productSearchHit.variant_meta_data.eu.price}
      maxPrice={productSearchHit.variant_meta_data.eu.price}
      productPath={productSearchHit.product_path}
      productRating={productSearchHit.product_rating}
      imageHeight={productSearchHit.image_height}
      imageWidth={productSearchHit.image_width}
    />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('.c-product-price')).toIncludeText(`Â£${productSearchHit.variant_meta_data.eu.price}`)
  expect(wrapper.find('.c-product-listing-card__rating')).toHaveClassName('c-product-listing-card__rating')

  // Assert lazyload component is loading image
  expect(wrapper.find('.lazyload-placeholder')).toHaveClassName('lazyload-placeholder')
})
