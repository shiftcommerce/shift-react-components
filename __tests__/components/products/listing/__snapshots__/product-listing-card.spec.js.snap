// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProductListingCard correctly 1`] = `
<ProductListingCard
  assetFileAltText="SEPIA nonslip tray 420×210mm"
  assetFileUrl="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg"
  maxPrice={97.68}
  minPrice={97.68}
  productPath="/products/seed_product_51"
  productRating={0}
  title="SEPIA nonslip tray 420×210mm"
>
  <div
    className="c-product-listing-card"
  >
    <div
      className="c-product-listing-card__body"
    >
      <Link
        as="/products/seed_product_51"
        href="/slug?slug=/products/seed_product_51"
      >
        <a
          as="/products/seed_product_51"
          href="/slug?slug=/products/seed_product_51"
        >
          <div
            className="c-product-listing-card__gallery o-image"
          >
            <LazyLoad
              alt="SEPIA nonslip tray 420×210mm"
              aria-label="SEPIA nonslip tray 420×210mm"
              className="c-product-listing-card__image u-image-shadow"
              imageHeight={280}
              imageWidth={280}
              src="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg"
            >
              <div
                style={
                  Object {
                    "backgroundColor": "#EEEFF3",
                    "height": 0,
                    "paddingBottom": "100.00%",
                    "width": "100%",
                  }
                }
              >
                <FadeIn
                  offset={150}
                  once={true}
                  render={[Function]}
                >
                  <LazyLoad
                    offset={150}
                    once={true}
                    overflow={false}
                    resize={false}
                    scroll={true}
                    unmountIfInvisible={false}
                  >
                    <div
                      className="lazyload-placeholder"
                      style={
                        Object {
                          "height": undefined,
                        }
                      }
                    />
                  </LazyLoad>
                </FadeIn>
              </div>
            </LazyLoad>
          </div>
          <div
            className="c-product-listing-card__title"
          >
            <p
              aria-level="1"
              role="heading"
            >
              SEPIA nonslip tray 420×210mm
            </p>
          </div>
        </a>
      </Link>
      <div
        className="c-product-listing-card__price"
      >
        <ProductPrice
          maxPrice={97.68}
          minPrice={97.68}
        >
          <div
            className="c-product-price"
          >
            <span>
              £
              97.68
            </span>
          </div>
        </ProductPrice>
      </div>
      <div
        className="c-product-listing-card__rating"
      >
        <Rating
          rating={0}
        >
          <span
            className="o-rating"
            key="0"
          >
            <span
              className="o-rating__star o-rating__star--empty"
              key="1"
            >
              ★
            </span>
            <span
              className="o-rating__star o-rating__star--empty"
              key="2"
            >
              ★
            </span>
            <span
              className="o-rating__star o-rating__star--empty"
              key="3"
            >
              ★
            </span>
            <span
              className="o-rating__star o-rating__star--empty"
              key="4"
            >
              ★
            </span>
            <span
              className="o-rating__star o-rating__star--empty"
              key="5"
            >
              ★
            </span>
          </span>
        </Rating>
      </div>
    </div>
  </div>
</ProductListingCard>
`;
